src/walletd/runtime.rs

```diff
@@ -360,8 +360,18 @@ impl Runtime {
                 error!("Msg and source don't have same swap_id, ignoring...");
                 return Ok(());
             }
+
+            Request::Protocol(msg)
+                if req_swap_id.is_some() && Some(msg.swap_id()) == req_swap_id => {}
             // TODO enter farcasterd messages allowed
-            _ => {}
+            _ if source == ServiceId::Farcasterd => {}
+            _ => {
+                error!(
```

Comment:
Can you move the two error arms so that they follow one another? That would make it easier to parse.